From 6285933470bab2923e4573b5d54ba9a32629b0cd Mon Sep 17 00:00:00 2001
From: Garvin Hicking <blog@garv.in>
Date: Thu, 12 Jan 2017 12:02:27 +0100
Subject: [PATCH]     * [Security] Redirection of comment.php now checks the
 referrer       and only allows the blog's host (thanks to Lee Sheldon Victor)

---
 comment.php | 2 +-
 docs/NEWS   | 3 +++
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/comment.php b/comment.php
index b152a7640..b43871aa1 100644
--- a/comment.php
+++ b/comment.php
@@ -11,7 +11,7 @@
 
 if (isset($serendipity['GET']['delete'], $serendipity['GET']['entry'], $serendipity['GET']['type'])) {
     serendipity_deleteComment($serendipity['GET']['delete'], $serendipity['GET']['entry'], $serendipity['GET']['type']);
-    if (serendipity_isResponseClean($_SERVER['HTTP_REFERER'])) {
+    if (serendipity_isResponseClean($_SERVER['HTTP_REFERER']) && preg_match('@^https?://' . preg_quote($_SERVER['HTTP_HOST'], '@') . '@imsU')) {
         header('Status: 302 Found');
         header('Location: '. $_SERVER['HTTP_REFERER']);
         exit;
diff --git a/docs/NEWS b/docs/NEWS
index 171310e74..1cba2d217 100644
--- a/docs/NEWS
+++ b/docs/NEWS
@@ -17,6 +17,9 @@ Version 2.1 ()
       
     * [Security] Reject %0D/%0A in exit tracking and other places
       (Issue #434)
+      
+    * [Security] Redirection of comment.php now checks the referrer
+      and only allows the blog's host (thanks to Lee Sheldon Victor)
 
     * Disabled Selenium test files unless enabled
 
